src/ORM/AssociationCollection.php

```diff
@@ -55,6 +57,30 @@ public function add($alias, Association $association)
     }
 
     /**
+     * Creates and adds the Association object to this collection.
+     *
+     * @param string $className The name of association class.
+     * @param string $associated The alias for the target table.
+     * @param array $options List of options to configure the association definition.
+     * @return \Cake\ORM\Association
+     * @throws InvalidArgumentException
+     */
+    public function load($className, $associated, array $options = [])
+    {
+        $options += [
+            'tableLocator' => $this->getTableLocator()
+        ];
+
+        $association = new $className($associated, $options);
```

Comment:
Now that you pass the locator here, it is trivial to let the tables pass the locator when building the `AssociationCollection`